### DISCUSSION TITLE
Version 2.1.2

### DISCUSSION THREAD

**Discussion by guydavis** [guydavis/machinaris#949] [Announcements]
### Added 
 - Settings | Plotting | Bladebit now supports "no_direct_io: true" option
 - Settings | Plotting | Bladebit now supports "check_plots: 100" option to check at end of plot creation
 - Settings | Plotting | Bladebit now supports "check_threshold: 0.6" option to delete invalid plots at end of plot creation
### Changed
 - Support for [Gigahorse Recompute Server](https://github.com/guydavis/machinaris/wiki/Gigahorse#recompute-server) (single GPU on fullnode/farmer), shared by multiple Gigahorse harvesters.
### Updated
 - [Chia](https://github.com/Chia-Network/chia-blockchain/releases/tag/2.1.2) to v2.1.2 - misc improvements, see their release notes.
 - [Gigahorse](https://github.com/madMAx43v3r/chia-gigahorse/releases/tag/v2.1.1.giga25) to v2.1.1.giga25 with C20 compression support.
 - [Flask](https://flask.palletsprojects.com/en/3.0.x/changes/#version-3-0-0) to v3.0.0 - serves up the Machinaris WebUI.
### Notes
 - Regression in Chia binaries (since v2.0.0) ignoring the "-y" parameter on plotnft changes, has broken pool join/leave thru WebUI.

<hr /><em>This discussion was created from the release <a href='https://github.com/guydavis/machinaris/releases/tag/2.1.2'>Version 2.1.2</a>.</em>

**Comment by shangyexin** [comment]
After I upgraded to the latest version I got the error below and now I can only roll back to v2.1.0.
`/machinaris/scripts/start_machinaris.sh: line 65: /chia-blockchain/venv/bin/gunicorn: No such file or directory`

**Reply by guydavis** [reply]
Sorry, it seems CNI changed their repository branch names yesterday, right in the middle of my builds.  I've now adopted their new naming standard and rebuilt the Machinaris v2.1.2 Docker image.  Please re-pull for `:latest`.  Sorry for the trouble and thanks for the report.
